File: domain/src/main/java/com/study/platform/domain/team/model/TeamScheduleRepository.java
package com.study.platform.domain.team.model;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface TeamScheduleRepository {

    TeamSchedule save(TeamSchedule schedule);
    Optional<TeamSchedule> findById(UUID id);
    void delete(TeamSchedule schedule);
    List<TeamSchedule> findAllByScheduledAtBetweenWithTeam(LocalDateTime start, LocalDateTime end);
}

File: domain/src/main/kotlin/com/study/platform/domain/team/model/TeamScheduleRepository.kt
package com.study.platform.domain.team.model

import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice
import java.time.LocalDateTime
import java.util.UUID

interface TeamScheduleRepository {

    fun save(schedule: TeamSchedule): TeamSchedule
    fun findById(id: UUID): TeamSchedule?
    fun delete(schedule: TeamSchedule)
    fun deleteAllByTeamId(teamId: UUID)
    fun findAllByScheduledAtBetweenWithTeam(start: LocalDateTime, end: LocalDateTime, pageable: Pageable): Slice<TeamSchedule>
}
